test/unit/org/apache/cassandra/index/sai/cql/DropIndexWhileQueryingTest.java

```diff
@@ -0,0 +1,104 @@
+                                                       ".dropIndexForBytemanInjections" +
+                                                       "(\"DROP INDEX cql_test_keyspace." + indexName + "\")"))
+                                        .build();
+
+        Injections.inject(injection);
+        injection.enable();
+        assertTrue("Injection should be enabled", injection.isEnabled());
+        execute("INSERT INTO %s (electronicVehicle, modelYear, make, vehType) VALUES (?, ?, ?, ?)", "car", 1998, "Ford", "SUV");
+        assertInvalidMessage("An index may have been dropped.Cannot execute this query as it might involve " +
+                             "data filtering and thus may have unpredictable performance. If you want to execute this query" +
+                             " despite the performance unpredictability, use ALLOW FILTERING",
+                             "SELECT * FROM %s WHERE modelYear IN (1998, 2000) OR (make IN ('FORD', 'OPEL' ) OR vehType IN ('car', 'truck'));");
+    }
+
+    @Test
+    public void testReproductionForCNDB10535() throws Throwable
+    {
+        createTable("CREATE TABLE %s (pk int, str_val text, val vector<float, 3>, PRIMARY KEY(pk))");
+        String indexName = createIndex("CREATE CUSTOM INDEX ON %s(val) USING 'StorageAttachedIndex'");
+        waitForTableIndexesQueryable();
+
+        Injection injection = Injections.newCustom("drop_index2")
+                                        .add(newInvokePoint()
+                                             .onClass(QueryController.class)
+                                             .onMethod("buildPlan")
+                                             .atExit())
+                                        .add(ActionBuilder
+                                             .newActionBuilder()
+                                             .actions()
+                                             .doAction("org.apache.cassandra.index.sai.cql.DropIndexWhileQueryingTest" +
+                                                       ".dropIndexForBytemanInjections" +
+                                                       "(\"DROP INDEX cql_test_keyspace." + indexName + "\")"))
+                                        .build();
+
+        Injections.inject(injection);
+        injection.enable();
+        assertTrue("Injection should be enabled", injection.isEnabled());
+
+        execute("INSERT INTO %s (pk, str_val, val) VALUES (0, 'A', [1.0, 2.0, 3.0])");
+        assertInvalidMessage("An index may have been dropped. Ordering on non-clustering column requires the " +
+                             "column to be indexed", "SELECT pk FROM %s ORDER BY val ann of [0.5, 1.5, 2.5] LIMIT 2");
+    }
+
+    // the method is used by the byteman rule to drop the index
+    public static void dropIndexForBytemanInjections(String query)
+    {
+        String fullQuery = String.format(query, KEYSPACE);
```

Comment:
We set the keyspace here and in the injected calls, one is redundant. We could make this method take only the index name:
```
// the method is used by the byteman rule to drop the index
@SuppressWarnings("unused")
public static void dropIndexForBytemanInjections(String indexName)
{
    String fullQuery = String.format("DROP INDEX %s.%s", KEYSPACE, indexName);
    logger.info(fullQuery);
    schemaChange(fullQuery);
}
```

Reply:
Ops missed that one, one moment